### PR TITLE
Add function to create replacement map based on input wire expressions.

### PR DESCRIPTION
**Pull request by thiskappaisgrey**
Added a function that pattern matches an expression against:
```
(Var "i_name" 1)
```
and create a replacement map:
```
[ (( Var "i_name" 1), i_name) ]
```

### CODE REVIEW

File: lakeroad-egglog/tests/egglog_tests.rs [hunk 1]
```diff
@@ -113,6 +113,43 @@ fn create_rewrites(
     for expr in &exprs {
         println!("{}", expr);
     }
+    /// Given an expression, like `(Var "i_out" 1)` - create a replacement map that replaces that with `i_out`
+    /// You can pass the replacement map returned by this function to replace_in_expr
```
Comment (gussmith23) [inline lakeroad-egglog/tests/egglog_tests.rs:117]: Could you add a [doctest](https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html) to this?

e.g.
```suggestion
    /// Given an expression, like `(Var "i_out" 1)` - create a replacement map that replaces that with `i_out`
    /// You can pass the replacement map returned by this function to replace_in_expr
    /// ```
    /// <some code that shows what this function does, with assert statements to check the output>
    /// ```
```

**Review by gussmith23 — CHANGES_REQUESTED** [review]

**Comment by gussmith23** [comment]
Thank you @thiskappaisgrey ! Just requested a bit more documentation.

**Comment by thiskappaisgrey** [comment]
Added documentation. It doesn't run but you can add it to an example test maybe..? @gussmith23

**Review by gussmith23 — APPROVED** [review]
Thanks @thiskappaisgrey !

**Comment by gussmith23** [comment]
Sorry, I didn't realize the function wasn't at the top level, so of course doctests didn't work. My misunderstanding!

**Pull request merged**